tests/mock-snapd.c

```diff
@@ -796,6 +796,22 @@ mock_snapd_add_change (MockSnapd *self)
     return add_change (self);
 }
 
+void
+mock_change_set_kind (MockChange *change, const gchar *kind)
+{
+    g_free (change->kind);
+    change->kind = g_strdup (kind);
+}
+
+void
+mock_change_add_data (MockChange *change, const JsonNode *data)
+{
+    if (change->data != NULL) {
+        json_node_unref (change->data);
+    }
+    change->data = (JsonNode *)data;
```

Comment:
This would be less error prone with json_node_ref (don't forget to unref in the calling function).